sklearn/model_selection/_search.py

```diff
@@ -551,41 +552,78 @@ def _fit(self, X, y, labels, parameter_iterable):
             pre_dispatch=pre_dispatch
         )(delayed(_fit_and_score)(clone(base_estimator), X, y, self.scorer_,
                                   train, test, self.verbose, parameters,
-                                  self.fit_params, return_parameters=True,
+                                  self.fit_params,
+                                  return_train_score=self.return_train_score,
+                                  return_parameters=True,
                                   error_score=self.error_score)
           for parameters in parameter_iterable
           for train, test in cv.split(X, y, labels))
 
-        test_scores, test_sample_counts, _, parameters = zip(*out)
+        # if one choose to see train score, "out" will contain train score info
+        if self.return_train_score:
+            train_scores, test_scores, test_sample_counts, time, parameters =\
+                zip(*out)
+        else:
+            test_scores, test_sample_counts, time, parameters = zip(*out)
 
         candidate_params = parameters[::n_splits]
         n_candidates = len(candidate_params)
 
+        # if one choose to return train score, reshape the train_scores array
+        if self.return_train_score:
+            train_scores = np.array(train_scores,
+                                    dtype=np.float64).reshape(n_candidates,
+                                                              n_splits)
         test_scores = np.array(test_scores,
                                dtype=np.float64).reshape(n_candidates,
                                                          n_splits)
         # NOTE test_sample counts (weights) remain the same for all candidates
         test_sample_counts = np.array(test_sample_counts[:n_splits],
                                       dtype=np.int)
 
-        # Computed the (weighted) mean and std for all the candidates
+        # Computed the (weighted) mean and std for test scores
         weights = test_sample_counts if self.iid else None
-        means = np.average(test_scores, axis=1, weights=weights)
-        stds = np.sqrt(np.average((test_scores - means[:, np.newaxis]) ** 2,
-                                  axis=1, weights=weights))
+        test_means = np.average(test_scores, axis=1, weights=weights)
+        test_stds = np.sqrt(
+            np.average((test_scores - test_means[:, np.newaxis]) ** 2, axis=1,
+                       weights=weights))
+
+        time = np.array(time, dtype=np.float64).reshape(n_candidates, n_splits)
+        time_means = np.average(time, axis=1)
```

Comment:
It's triplication only if you would be okay with taking the weighted averaging for all the test scores, train scores and times...